python/runtime/cudaq/platform/py_alt_launch_kernel.cpp

```diff
@@ -561,8 +561,8 @@ MlirModule synthesizeKernel(const std::string &name, MlirModule module,
   pm.addNestedPass<func::FuncOp>(cudaq::opt::createExpandMeasurementsPass());
   pm.addNestedPass<func::FuncOp>(cudaq::opt::createClassicalMemToReg());
   pm.addNestedPass<func::FuncOp>(createCanonicalizerPass());
-  pm.addNestedPass<func::FuncOp>(cudaq::opt::createLoopNormalize());
-  pm.addNestedPass<func::FuncOp>(cudaq::opt::createLoopUnroll());
+  pm.addPass(cudaq::opt::createLoopNormalize());
+  pm.addPass(cudaq::opt::createLoopUnroll());
```

Comment:
Sorry, I wasn't precise enough. Let's leave these as Func level passes so we can utilize multithreading. There is no need to run them at the module level AFAICT.

Reply:
The loop unroll complains if I make it a func pass: `error: can't run 'builtin.module' pass manager on 'func.func' op`

Reply:
OK, interesting. I'll take a closer look at it.

Reply:
There's a bug in loop-unroll that prevents the pass from being run at a FuncOp level. I'll create a PR to fix it.

Reply:
See #2422

Reply:
would it be possible to submit it after this PR to speed things up?

Reply:
If this PR is merged first, please open an issue for this discussion so someone can follow up on the technical debt that isn't addressed here.

Reply:
nevermind:)